File: DependencyInjection/DependencyInjection/_1_ControlFreak/ApplicationRoot.cs
using DependencyInjection._1_ControlFreak.Core;

namespace DependencyInjection._1_ControlFreak
{
    public class ApplicationRoot
    {
      public void Main()
      {
        var sys = new TeleComSystem();

        sys.Start();
      }

      public void Main_IoC_Container()
      {
        //TODO mention RRR pattern

        var sys = new TeleComSystem();

        sys.Start();
      }

    }

}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/ApplicationRoot.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Core.TeleComSystem;

public class ApplicationRoot {
      public void main() {
        TeleComSystem sys = new TeleComSystem();

        sys.start();
      }

      public void mainIoCContainer() {
        //TODO add container here during the training
        //TODO mention RRR pattern

        TeleComSystem sys = new TeleComSystem();

        sys.start();
      }

    }